lib/mongoose_push_web/plugs/cast_and_validate.ex

```diff
@@ -0,0 +1,80 @@
+defmodule MongoosePushWeb.Plug.CastAndValidate do
+  @moduledoc """
+  Module plug that serves as a wrapper for OpenApiSpex.Plug.CastAndValidate plug,
+  to overcome difficulties with proper message validating. For more details,
+  please refer to update_schema_and_do_call/2 function comment.
+  """
+  @behaviour Plug
+
+  @impl Plug
+  def init(opts) do
+    OpenApiSpex.Plug.CastAndValidate.init(opts)
+  end
+
+  @impl Plug
+  def call(conn, opts) do
+    stub_conn = %Plug.Conn{
+      conn
+      | adapter: {MongoosePushWeb.Plug.CastAndValidateStubAdapter, %{}}
```

Comment:
How about:
```suggestion
      | adapter: {MongoosePushWeb.Plug.CastAndValidate.StubAdapter, %{}}
```
with the file rename as well ;)

Reply:
@rslota,

You got me, I just was too lazy to create a separate folder for this file, what I will make up in the upcoming PR. Thanks again for your engagement, obviously I will wait for guys to approve this before we merge :)